### DISCUSSION TITLE
Photos lost!

### DISCUSSION THREAD

**Discussion by gk1089** [photoprism/photoprism#2298] [Q&A]
After a lot of research for web-based photo repository, I zeroed on Photoprism early this year to manage the old, scanned photos of our office. I set it up in a Virtual Machine with following specifications:

OS: Ubuntu 20.04.03
CPU : 2 Cores
RAM: 8 GB
HDD: 200 GB
Photoprism version: 211215-93b26f19-Linux-x86_64 installed using Docker

All was fine upon installation and I uploaded more than 2000 photos and painstakingly added dates to them (which were missing since they were scanned), grouped them into albums and finally updated the facial information (my main reason to choose Photoprism). Since a lot of photos were still remaining to be uploaded I had not taken any snapshots of my VM and that turned out to be my cardinal sin.

One fine day, while trying to search and download photos of a person whose faces I had tagged earlier, I began getting errors and photos began getting deleted. It did not stop at just the photos of the person and in some time all the photos were gone. I got so frustrated that I abandoned it all these months in the hope that some such bug would surface in Photoprism community.

The complete log is available at [logs.txt](https://github.com/photoprism/photoprism/files/8601558/logs.txt)

Is it possible to set things right and get back to when things were just right with my Photoprism setup? If not, I have the raw photos available with me separately. Is it at least possible to recover the date/time tags and the facial information and any way to upload them separately.

I can provide any further details, logs, command output if it can help in pinning the issue down.

Thanks!

**Comment by graciousgrey** [comment]
PhotoPrism does not delete files from your originals folder unless you delete them using the web UI. 
At the physical location of your originals folder all files should still be present.

Your logs sound like PhotoPrism could not access the originals directory anymore. PhotoPrism detects changes you do on your filesystem, so that the index can be updated when files are moved or deleted.
If PhotoPrism is indexing and has no access to the originals directory it thinks all files have been removed and updates it's index accordingly. So you need to make sure the originals directory is always accessible to PhotoPrism when indexing.

To get your old state back you can try:
1. Make the originals folder accessible again/Find out what was the issue
2. Trigger a re-index

In case you did not run the purge or cleanup commands manually. This should recreate your index including the changes you did manually, such as adding dates.

**Reply by gk1089** [reply]
Thank you for your quick response! I have a folder called 'Pictures' in the same home directory where 'photoprism' (this is where I run the "docker-compose -d up" command upon boot) folder is located. This 'Pictures' folder contains my photos  grouped into folder by years. I am guessing this is where my photos went after I uploaded through the Web UI. Some photoprism online help pages talk about 'Originals' folder but I don't have any such in my set up.

Anyway, since I might have messed around in trying to install some other libraries, I modified the folder permission of the 'Pictures' folder to 755 and all the files recursively to 777 (just to see if it will work). I saved my VM state, restarted and hit scan upon restarting photoprism. 

But, it does not seem to have helped. The last 500 lines of the log after running rescan can be found in the file [log2.txt](https://github.com/photoprism/photoprism/files/8602053/log2.txt)

Could you please help me further identify the issue? Is there someplace I can modify and force photoprism to see the photos in 'Pictures' folder?

Thanks again.

**Reply by graciousgrey** [reply]
In general PhotoPrism offers two methods of adding files. Indexing and importing (https://docs.photoprism.app/user-guide/library/). The web upload uses the import method.

PhotoPrism's originals directory, is configured in your docker-compose.yml (https://docs.photoprism.app/getting-started/docker-compose/#photoprismoriginals). In our example docker-compose.yml files the originals directory is mounted to `~/Pictures`.

If the web upload moved files into year/month folders insides of your Pictures folder it was at least at this point configured as originals directory. Did you change something regarding the volume mounts in your docker-compose.yml?

You can follow this troubleshooting guide: https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions to ensure your folder has the right permissions.

**Reply by gk1089** [reply]
Thank you again for your reply. 

I have not changed anything in the yml file as seen in the attached file (saved with txt extension)
[docker-compose.txt](https://github.com/photoprism/photoprism/files/8619641/docker-compose.txt)

I have two relevant folders in the "/home/photoadmin" of my Virtual Machine: **photoprism** (containing the docker files) and **Pictures** (containing the originally uploaded photos. I made a file listing of both the folders using 'tree' utility, if that will help me resolve the issue in any way.

- [photoprismtree.txt](https://github.com/photoprism/photoprism/files/8619701/photoprismtree.txt)
- [Picturestree.txt](https://github.com/photoprism/photoprism/files/8619711/Picturestree.txt)

As regards file/folder permissions, that does not seem to be the issue. As suggested in the troubleshooting guide linked by you, I changed file/folder permissions to both of the previously mentioned folders/files using the **sudo chmod -R a+rwx** command and ran rescan in photoprism. The process completed in 12 seconds but my pictures still did not show up.

Is there anything else I can do at my end? 

I sincerely thank you for your patience again.

**Reply by lastzero** [reply]
Just to be sure, did you also check permission to the storage folder or only originals? What kind of virtual machine is it? From what I understand, you have multiple layers of virtualization there, which exceeds the scope of our documentation and also the resources the core team has available for providing community support.

**Reply by gk1089** [reply]
Out of sheer frustration, I had shut the VM down. Your comment made me fire it up again and What-Do-You-Know, the thing is finally able to see the pictures. It is indexing currently and I hope all will be well now onwards. So, restarting after permission change somehow did the trick. I had already restarted the docker, but that did not help.

Thank you very much for your help! Just point me in the right direction as to what the original permissions for the file/folders should be so that I can revert them with due caution. The link that you had provided above only talked about removing restrictions and not adding them.

Thanks indeed!
:-)

**Comment by aldobranti** [comment]
It seems to me that deleting the file from PhotoPrism's database and world view ( accepted that the image remains untouched in Originals) is just plain heavy handed. It loses the effort made by your user to categorise and establish an album level view of the user's images.
So my proposal is: when the file is not found in its previously indexed position, it can be marked for non display. If the same file is later found in a different position in the Originals file system -- where same file is identified by its sum check -- the indexing can be updated and  the image becomes viewable again together with its previously assigned membership of albums etc. Thus the user's effort is retrieved.
What is the risk that two totally different images have the same hash? -- personally I will live with that, and the interest in finding such a case.
So what about files that the user actually removes from Originals? A list of these with their last known indexed position in Originals can be added to the sidebar and the user should review and make their own individual choices for these
And what about the storage hit ? The data cost of hanging onto a hash value and sidecar bits will be a fraction of the overall storage demand of the average user's data footprint. The proposed retention system might be disabled for specific users who have a workflow with very high turnover of images in the originals areas -- many more users rely on PhotoPrism to index a much more stationary population of images,

**Comment by lastzero** [comment]
> when the file is not found in its previously indexed position, it can be marked for non display. If the same file is later found in a different position in the Originals file system -- where same file is identified by its sum check -- the indexing can be updated and the image becomes viewable again together with its previously assigned membership of albums etc.

This should already be the case, unless the SHA1 checksum of the file has been changed by editing the file. In this case, there is no way to determine if two files are the same if the filename and/or path have also changed.

Unfortunately, we don't have the resources to investigate every reported problem in depth, otherwise I could give you more specific reasons as to what might have gone wrong in your case.

**Comment by aldobranti** [comment]
So, when a change to prevent indexing if Originals is offline gets implemented, that which I have described will be effective Bring it on and thank you for Photoprism All best Aldobranti -- aldobranti.org, a performance 
-------- Original message --------From: Michael Mayer ***@***.***> Date: 25/07/2022  13:45  (GMT+00:00) To: photoprism/photoprism ***@***.***> Cc: aldobranti ***@***.***>, Comment ***@***.***> Subject: Re: [photoprism/photoprism] Photos lost! (Discussion #2298) 

when the file is not found in its previously indexed position, it can be marked for non display. If the same file is later found in a different position in the Originals file system -- where same file is identified by its sum check -- the indexing can be updated and the image becomes viewable again together with its previously assigned membership of albums etc.

This should already be the case, unless the SHA1 checksum of the file has been changed by editing the file. In this case, there is no way to determine if two files are the same if the filename and/or path have also changed.
Unfortunately, we don't have the resources to investigate every reported problem in depth, otherwise I could give you more specific reasons as to what might have gone wrong in your case.
